xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-requiredrights/xwiki-platform-security-requiredrights-api/src/main/java/org/xwiki/platform/security/requiredrights/RequiredRightAnalysisResult.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.platform.security.requiredrights;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.stability.Unstable;
+import org.xwiki.text.XWikiToStringBuilder;
+
+/**
+ * Represents the result of a required right analysis. This class provides getters and setters for various properties
+ * related to the analysis result.
+ *
+ * @version $Id$
+ * @since 15.9RC1
+ */
+@Unstable
+public class RequiredRightAnalysisResult
+{
+    private EntityReference entityReference;
```

Comment:
It feels weird that the entity reference is the only field not final here... I would almost have assumed the contrary.

Reply:
I think we can now make it final, we had some analyzers before that didn't set the entity reference as they didn't know it and the idea was to make it easy to set it later at a different place in the code (think, e.g., macro analyzers being called from an XObject analyzer). The code has been changed in the meantime to ensure that all analyzers know the entity reference.

Reply:
Confirmed, `setEntityReference` is currently unused, making it final.

Reply:
Now honestly I'd probably only have kept `entityReference` and `requiredRight` final, and I'd have put setters for the others.